app/client/cli/helpers/setup.go

```diff
@@ -27,24 +27,27 @@ func P2PDependenciesPreRunE(cmd *cobra.Command, _ []string) error {
 	bus := runtimeMgr.GetBus()
 	SetValueInCLIContext(cmd, BusCLICtxKey, bus)
 
-	setupPeerstoreProvider(*runtimeMgr, flags.RemoteCLIURL)
+	if err := setupPeerstoreProvider(*runtimeMgr, flags.RemoteCLIURL); err != nil {
+		return err
+	}
 	setupCurrentHeightProvider(*runtimeMgr, flags.RemoteCLIURL)
 	setupAndStartP2PModule(*runtimeMgr)
 
 	return nil
 }
 
-func setupPeerstoreProvider(rm runtime.Manager, rpcURL string) {
-	bus := rm.GetBus()
-	modulesRegistry := bus.GetModulesRegistry()
-	pstoreProvider := rpcPSP.Create(
-		rpcPSP.WithP2PConfig(rm.GetConfig().P2P),
-		rpcPSP.WithCustomRPCURL(rpcURL),
-	)
-	modulesRegistry.RegisterModule(pstoreProvider)
+func setupPeerstoreProvider(rm runtime.Manager, rpcURL string) error {
+	// Ensure `PeerstoreProvider` exists in the modules registry.
+	_, err := rpcPSP.Create(rm.GetBus(), rpcPSP.WithCustomRPCURL(rpcURL))
+	if err != nil {
```

Comment:
As a single line, this is 90 characters long which is ~10 characters out of my comfort zone; however, since it's clear that the err is the only thing introduced in the new scope and it's conventionally obvious what's going on, I have no principled objection. Just felt the need to share the feeling, the suggestion induced. :sweat_smile: